packages/eslint-plugin/tests/rules/member-ordering-alphabetically-order.test.ts

```diff
@@ -213,6 +225,17 @@ class Foo {
         { default: { memberTypes: defaultOrder, order: 'alphabetically' } },
       ],
     },
+
+    // default option + enum with specified values + lower/upper case
+    {
+      code: `
+enum Foo {
+  a = 0,
+  A = 1,
+}
+            `,
+      options: [{ default: { memberTypes: 'never', order: 'alphabetically' } }],
+    },
```

Comment:
We'll need a few more test cases to verify that enums work the same as other constructs. At least:

* `alphabetical-case-insensitive` and `"as-written"` sorting
* Enums with intermixed implicit values and/or string values
* Computed property names (see above)

Reply:
I added a few more tests, I'm not sure if this covers everything you are requesting though, as I am not super familiar with this rule.

I will probably need help with implementing any remaining outstanding test cases, as well as https://github.com/typescript-eslint/typescript-eslint/issues/4859#issuecomment-1109022523

Reply:
A good start! A few more cases to consider...

```ts
// What I mean by intermixed implicit values
enum Foo {
    a,
    b = 1,
    c,
    e,
    d
}
```

```ts
// What I mean by string values
enum Foo {
    a,
    b = 'x'
}
```

```ts
// TypeScript actually allows this...
enum Foo {
    a,
    b = 'x',
    c = 2
}
```

```ts
// ...and this!
enum Foo {
    a,
    b = '',
    c = 2,
    d
}
```

Additionally, the computed cases I mentioned in the other comment should all have tests too. `['a']` and `['a' + 'b']` and the like.